worker/api.js

```diff
@@ -0,0 +1,40 @@
+export async function handleAPIRequest(request) {
+  const url = new URL(request.url);
+  const accept = request.headers.get("accept");
+  const isJSON = accept && accept.indexOf("application/json") >= 0;
+
+  if (!isJSON) {
+    return new Response(
+      "The client does not accept 'application/json' content",
+      {
+        status: 400,
+        statusText: "Bad Request"
+      }
+    );
+  }
+
+  if (url.pathname == "/api/versions") {
```

Comment:
Optionally, you could check whether the request method is `GET` (or `HEAD`, or `OPTIONS`) and, if not, return `405 Method Not Allowed`.

If the method is `HEAD`, you could return the headers only, without the body.

If the method is `OPTIONS`, you could list allowed request methods by responding with an `Allow: <methods>` header and handle CORS headers so this API can be called from a client in the browser.